### PR TITLE
Show nice error message instead of crashing if version incrementing commits are missing

### LINKED ISSUE #65: Error: Cannot read property 'trim' of null
Attempting to run `versionist` without any arguments in the `versinist` repo (commit 7862dd5) produces the following error:

```
Cannot read property 'trim' of null
TypeError: Cannot read property 'trim' of null
    at Function.clean (/usr/local/lib/node_modules/versionist/node_modules/semver/semver.js:266:24)
    at Object.module.exports.updateVersion.npm (/usr/local/lib/node_modules/versionist/lib/presets.js:271:37)
    at apply (/usr/local/lib/node_modules/versionist/node_modules/lodash/lodash.js:499:17)
    at Object.wrapper [as updateVersion] (/usr/local/lib/node_modules/versionist/node_modules/lodash/lodash.js:5383:16)
    at async.waterfall (/usr/local/lib/node_modules/versionist/bin/cli.js:178:26)
    at nextTask (/usr/local/lib/node_modules/versionist/node_modules/async/dist/async.js:5070:14)
    at /usr/local/lib/node_modules/versionist/node_modules/async/dist/async.js:5064:13
    at apply (/usr/local/lib/node_modules/versionist/node_modules/async/dist/async.js:21:25)
    at /usr/local/lib/node_modules/versionist/node_modules/async/dist/async.js:56:12
    at /usr/local/lib/node_modules/versionist/node_modules/async/dist/async.js:840:16
```

This looks like it could be related to that commit missing a change-type entry, but still, it'd be nice if we handled this error case more gracefully

### PR DESCRIPTION
**Pull request by lekkas**
Closes #65

### CODE REVIEW

File: bin/cli.js [hunk 1]
```diff
@@ -154,7 +154,7 @@ async.waterfall([
 
     if (_.includes(documentedVersions, version)) {
       console.log(`Omitting: ${version}`);
```
Comment (craig-mulligan) [inline bin/cli.js]: Now that we have the debug package should we start moving `console.logs` to it? Or best to do it all at once?

Comment (lekkas) [inline bin/cli.js]: Good catch! Btw do we even need this 'Omitting' message here? if no versions are found `versionist` will exit anyway with the message that describes what went wrong.

**Comment by lekkas** [comment]
Added tests

File: tests/cli/cases/missing-changetype-commits.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  '  },',
+  '  template: [',
+  '    \'## {{version}}\',',
+  '    \'\',',
+  '    \'{{#each commits}}\',',
+  '    \'{{#with footer}}\',',
+  '    \'- {{capitalize Changelog-Entry}}\',',
+  '    \'{{/with}}\',',
+  '    \'{{/each}}\'',
+  '  ].join(\'\\n\')',
```
Comment (lurch) [inline tests/cli/cases/missing-changetype-commits.js]: Yuck! ;-)  I guess this would be more readable if it was simply read in from an external file?

Comment (lekkas) [inline tests/cli/cases/missing-changetype-commits.js]: I've followed @jviotti approach from the existing test cases in `tests/cli/cases/` .

We can definitely add a separate folder to keep external files (e.g. `fixtures/`) but if we want to refactor tests I'd vote for opening a separate issue/PR.

File: bin/cli.js (hunk 1 above)
Comment (craig-mulligan) [inline bin/cli.js]: ah I see, yea we don't need that.

**Pull request merged**